pulp_python/app/__init__.py

```diff
@@ -11,3 +11,10 @@ class PulpPythonPluginAppConfig(PulpPluginAppConfig):
     version = "3.12.0.dev"
     python_package_name = "pulp-python"
     domain_compatible = True
+
+    def import_viewsets(self):
+        """Ensure that the PyPI urls are included in Pulp's RBAC system"""
+        super().import_viewsets()
+        from pulp_python.app.pypi.views import PyPIView, SimpleView, UploadView, MetadataView
+
+        self.named_viewsets[None].extend([PyPIView, SimpleView, UploadView, MetadataView])
```

Comment:
This looks very hacky. Can you explain how this works? `None` as a dict key really tingles.

Reply:
It is hacky. The roles & access policies migrate hook use this dict to figure out which viewsets it needs to inspect. Both hooks use `values` to get the viewsets, so since the PyPI views don't have a backing model we can use `None` as the key. The big thing with putting non-`NamedModelViewSets` inside this dict is that you need to have your viewset mock one by implementing ~5 methods (the ones on the `PyPIMixin`) in order for other parts of Pulp to not break. 

I am not that pleased with this solution, it's pretty tenuous, but we currently don't have any way to register other views with our RBAC system.

Reply:
First, we should have a better mechanism.
Can't you just create the record in the db from a migration hook here?
Would we need a special PermissionClass to find the access policy without assuming it's a NamedModelVS?

Reply:
At the very least, we need a plan how it should look better. And a comment here so we know what this is actually accomplishing and when to rework it.

Reply:
Ok, I agree the hack is too much. I removed it and copied/pasted the access policy hook from pulpcore and added a comment to revisit once we have a better mechanism in pulpcore: https://github.com/pulp/pulpcore/issues/5500

Reply:
Thank you!